crates/matrix-sdk-base/src/store/integration_tests.rs

```diff
@@ -1729,3 +1765,23 @@ fn custom_presence_event(user_id: &UserId) -> Raw<PresenceEvent> {
 
     Raw::new(&ev_json).unwrap().cast()
 }
+
+fn custom_beacon_info_event(
+    user_id: &UserId,
+    live: bool,
+    duration_millis: u64,
+) -> Raw<AnySyncStateEvent> {
+    let content =
+        BeaconInfoEventContent::new(None, Duration::from_millis(duration_millis), live, None);
+
+    let event = json!({
+        "event_id": "$h29iv0s8:example.com",
+        "content": content,
+        "sender": user_id,
+        "type": "org.matrix.msc3672.beacon_info",
+        "origin_server_ts": 0u64,
+        "state_key": user_id,
+    });
+
+    serde_json::from_value(event).unwrap()
```

Comment:
Any chance you can use one of the `EventBuilder::make_state_event…` functions? It would avoid an extra adhoc creation of an event based on JSON.

Reply:
Unused now that Ive removed `test_beacon_saving()` from this file